crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -32,6 +33,44 @@ pub fn generate_merkle_tree_from_grouped_txs<T: AsRef<[u8]>>(
     tree
 }
 
+#[derive(Clone, Copy, Debug, thiserror::Error)]
+pub enum ChainIdsVerificationFailure {
+    #[error(
+        "the provided chain IDs root could not be verified against the provided proof and data \
+         hash/root"
+    )]
+    ChainIdsRootNotInData,
+    #[error(
+        "the provided chain IDs root does not match the root reconstructed from the provided \
+         chain IDs"
+    )]
+    ChainIdsRootDoesNotMatch,
+}
+
+/// Utility to check the validity of sequencer block fields related to chain IDs.
+///
+/// # Errors
+/// Returns one of the variants described in [`ChainIdsVerificationFailure`] as an error.
+pub fn assert_chain_ids_are_included<'a, TChainIds: 'a>(
```

Comment:
This function is now also use to construct `SequencerBlockData`.

I I also noticed that we were not ensuring that the chain ids commitment/root was checked, so it felt natural to also add that here. This adds to #548